Code/Editor/Lib/Tests/test_AzAssetBrowserRequestHandler.cpp

```diff
@@ -226,17 +226,17 @@ namespace UnitTest
                         {
                             if (comptype != AZ::Uuid(MockEditorComponent::s_MockEditorComponentTypeId))
                             {
-                                return ThisOutcome(AZStd::string("Failed - wrong component"));
```

Comment:
question: Would it be possible to move the `AZStd::unexpect` inside the `ThisOutcome` constructor so it's better hidden and fewer callsites need to be updated?

Reply:
I see, we still likely need to issue a deprecation notice around this though as this could break people. If the current code could be at risk of already being broken we might be able to argue making this change but otherwise for a fairly widely used type we're breaking the API without any notice period for people to upgrade without being broken

Reply:
I will add a deprecation notice for the API.
But based on a search of where AZ::Outcome is used across known git repositories using sourcegraph, it is only referenced in @AMZN-Olex book(https://sourcegraph.com/search?q=context:global+AZ::Outcome+NOT+repo:o3de&patternType=standard)

Reply:
Understood, thanks for adding issue #12696 @lumberyard-employee-dm 👍 Once this is merged is likely makes sense to send an impactful change notification to Discord too

Reply:
Will do.
I'll post it on Discord